network/ipmasq.go

```diff
@@ -57,3 +57,17 @@ func setupIPMasq(ipn ip.IP4Net) error {
 
 	return nil
 }
+
+func teardownIPMasq(ipn ip.IP4Net) error {
+	ipt, err := iptables.New()
+	if err != nil {
+		return fmt.Errorf("failed to setup IP Masquerade. iptables was not found")
+	}
+
+	err = ipt.DeleteChain("nat", "FLANNEL")
```

Comment:
Isn't a call to `ipt.ClearChain` required before `ipt.DeleteChain`. From the code comments [here] (https://github.com/coreos/go-iptables/blob/master/iptables/iptables.go) I understand that the `DeleteChain` requires that the chain already be empty.